core/src/miner/backup.rs

```diff
@@ -46,9 +87,9 @@ pub fn recover_to_data(db: &dyn KeyValueDB) -> HashMap<H256, MemPoolItem> {
     for (key, value) in db.iter(dblib::COL_MEMPOOL) {
         let bytes = (*value).to_vec();
         let rlp = rlp::Rlp::new(&bytes);
-
         let decoded_key = (key.as_ref()[PREFIX_SIZE..]).into();
-        let decoded_item = rlp.as_val().unwrap();
+        let mem_pool_projection: MemPoolItemProjection = rlp.as_val().unwrap();
+        let decoded_item = mem_pool_projection.try_into().expect("DB corruption failed");
```

Comment:
"DB corruption detected" seems more appropriate.

Reply:
I fixed it.